daily-rotate-file.js

```diff
@@ -137,9 +121,14 @@ if (semver.major(winston.version) === 2) {
     };
 }
 
-DailyRotateFile.prototype.close = function () {
+DailyRotateFile.prototype.end = function (chunk, encoding, callback) {
```

Comment:
I'm continuing to read about `Streams` -- I'm actually not sure `end()` should be overridden.  The [base implementation](https://github.com/nodejs/node/blob/9e9c516512668133813c71e39d56b011b804572b/lib/_stream_writable.js#L549-L575) does a lot more work which is not being handled here, including managing the corked state of the stream

This [trivial example](https://nodejs.org/api/stream.html#stream_decoding_buffers_in_a_writable_stream) for decoding within a stream also wraps another stream and they do their cleanup as part of `_destroy()`. I'm still not clear on the full lifecycle of Streams, so this is all still a little mysterious to me, but my gut says that overriding `end()` might have odd side effects later.

Reply:
Good point. Pushing a version that does the cleanup in `_final()`.